### PR TITLE
Config test backfill

### PR DESCRIPTION
**Pull request by dsabeti**
### What is this change about?

As we were working on adding a new test suite, we found that there were some inconsistencies in the CATs Config tests. These commits attempt to address the following issues:
- De-duping struct definitions by embedding structs rather than duplicating values
- Rename the `allConfig` test object to `nullConfig` because it exists specifically to test null values in the config json.
  - We considered trying to write a test to ensure that the nullConfig struct always matches the Config struct to prevent this divergence from occurring again in the future. We decided it was out of scope for what we were hoping to achieve today.
- Added assertions for untested null value conditions
- Added assertions for overriding `include_*` config parameters as these were previously untested
- Added assertions for default config values for missing `include_*` config parameters

### What version of cf-deployment have you run this cf-acceptance-test change against?
N/A (config test only)

### Please check all that apply for this PR:

- [ ] introduces a new test --- Are you sure everyone should be running this test?
- [ ] changes an existing test
- [ ] requires an update to a CATs integration-config

### Did you update the README as appropriate for this change?

- [ ] YES
- [x] N/A

### If you are introducing a new acceptance test, what is your rationale for including it CATs rather than your own acceptance test suite?

N/A

### How many more (or fewer) seconds of runtime will this change introduce to CATs?
N/A

### What is the level of urgency for publishing this change?

- [ ] **Urgent** - unblocks current or future work
- [x] **Slightly Less than Urgent**

### Tag your pair, your PM, and/or team!
@aramprice

### CODE REVIEW

**Comment by dsabeti** [comment]
@ctlong Just a quick note on merging this: I'd recommend _not_ squashing these commits when you get to merging this. The change is simple, but verbose, and I think it would be hard to follow if all the changes came through in a single commit.

File: helpers/config/config_struct.go [hunk 1]
```diff
@@ -686,6 +686,10 @@ func validateCredHubSettings(config *config) error {
 }
 
 func validateVolumeServices(config *config) error {
+	if config.IncludeVolumeServices == nil {
+		return nil
+	}
+
 	if !config.GetIncludeVolumeServices() {
```
Comment (ctlong) [inline helpers/config/config_struct.go:693]: I don't think that we need a `nil` check for both `config.IncludeVolumeServices` and `config.GetIncludeVolumeServices()`, as they should be pulling the same value.

**Review by ctlong — COMMENTED** [review]
@dsabeti @aramprice just one small comment. Otherwise it all LGTM!

Thanks!

File: helpers/config/config_struct.go (hunk 1 above)
Comment (aramprice) [reply helpers/config/config_struct.go:693]: I _think_ the issue is that `config.IncludeVolumeServices` is a pointer to a bool, so negating `config.GetIncludeVolumeServices()` will fail if the underlying pointer is `nil`.

The expression could probably be collapsed into something like:
`if config.GetIncludeVolumeServices() != nil && !config.GetIncludeVolumeServices() {`

Comment (ctlong) [reply helpers/config/config_struct.go:693]: Ok, if that's the case, then it's fine as is.
